src/components/progress-bar/components/TrackSegment.tsx

```diff
@@ -31,7 +31,7 @@ export const TrackSegment: FC<TrackSegmentProps> = ({
 	const colors = intersectedSegments.map(({ color }) => color);
 	// If there are no colors, it picks no color (undefined) = not the primary color.
 	const color = colors.length
-		? getHighlightColorBlended?.([...colors, defaultColor])
+		? getHighlightColorBlended?.([...colors, defaultColor].flat())
```

Comment:
In what scenario is this flat need? can either `color` or default `defaultColor` be an array>

Reply:
`defaultColor` - always string.
the issue is  with `colors` - line31 - which is `string[][]` because we set that `Highlight` has `color:string[]`, and a map of `Highlight[].colors` would be an array of strings arrays